### PR TITLE
fix: Adjusted typescript implementation for sdk

### PR DESCRIPTION
**Pull request by Danwakeem**
## Description
Changed the `index.d.ts` file to import the main implementation file via the `index.ts` file as well as export the `Sdk` type. This should give typescript projects typical intellisense and the ability to import the sdk like typical typescript packages.

### CODE REVIEW

File: node/packages/sdk/.ts-types/index.d.ts [hunk 1]
```diff
@@ -18,7 +19,7 @@ interface Sdk {
     options?: {
       startTime?: bigint;
       immediateDescendants?: string[];
-      tags?: Record<string, boolean | number | string | Date | Array | Null>;
+      tags?: Record<string, boolean | number | string | Date | Array<any> | null>;
```
Comment (Mmarzex) [inline node/packages/sdk/.ts-types/index.d.ts]: Probably would be better to use `Array<unknown>` here so it doesn't get caught by `tsc` that is configed for no any's.

**Review by Mmarzex — CHANGES_REQUESTED** [review]

File: node/packages/sdk/.ts-types/index.d.ts (hunk 1 above)
Comment (Danwakeem) [reply node/packages/sdk/.ts-types/index.d.ts]: Yeah that is a good call, made this change 😎

**Review by Mmarzex — APPROVED** [review]

**Comment by medikoo** [comment]
@Danwakeem this PR broke our extension builds.. all integration tests now fail. Do you know what could be the cause?

**Comment by Danwakeem** [comment]
@medikoo I pulled down the `main` branch, reinstalled dependencies, ran the build in the `aws-lambda-sdk` directory, and I am seeing the failed integration tests but I am not sure what exactly the error is? 🤔 

When I look a the content of the zip file that was created and look at the content of the `extension.internal/nodejs/node_modules/@serverless/sdk/index.js` I see some references to a seemingly local path and I am wondering if that may have something to do with it? 👇 

Screenshot 2023-01-11 at 11 40 03 AM

Does that look unusual to you? I am not sure what this looked like before but seems strange I guess. I also don't how or why this PR would have effected the builds. The build file seems to only reference the `index.js` file which is was unchanged here 🤔

**Comment by Danwakeem** [comment]
Yeah when I revert this PR those local paths seem to go back to normal 🤔 Weird. I am still playing around with this a bit.

**Comment by medikoo** [comment]
@Danwakeem I'm not sure as well what's going on. It appears ESbuild is trying to be smart with TS and something breaks.

Can we revert this PR, until this is figured out? I wanted to push new SDK release now, so we have it ready for tomorrow with recent changes

**Comment by Danwakeem** [comment]
@medikoo Yeah that is alright with me 👍 I can work on this in the background while you get the release out.

**Pull request merged**